other/coding/src/main/java/org/example/coding/loadbalancer/LoadBalancer.java

```diff
@@ -0,0 +1,60 @@
+package org.example.coding.loadbalancer;
+
+import org.example.coding.loadbalancer.model.Server;
+import org.example.coding.loadbalancer.pickstrategy.ServerPickStrategy;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class LoadBalancer {
+
+    private static final int MAX_CAPACITY = 10;
+
+    private final Map<String, Server> urlToServerMap = new ConcurrentHashMap<>();
+
+    private final ServerPickStrategy pickStrategy;
+
+    public LoadBalancer(ServerPickStrategy pickStrategy) {
+        if (pickStrategy == null) {
+            throw new IllegalArgumentException("Pick strategy cannot be null");
+        }
+        this.pickStrategy = pickStrategy;
+    }
+
+    public boolean registerServer(Server server) {
+        if (server == null) {
+            return false;
+        }
+
+        synchronized (this) {
```

Comment:
why synchronised if map is concurrent?

Reply:
There are several checks in sequence so there might be race conditions - so I think better to synchronise this piece of code and for adding server give access only to one thread

Reply:
Right, valid point, i think i'd try to use something like `putIfAbsent` to avoid doing separate contains check. But not sure if something similar would be possible with size. My point was it seems concurrent functions of concurrent hash map are not used here at all. Then also there's no sync in remove method, which i'm not sure might or might not be a problem. Overall looks suspicious :)